crates/xilem_core/src/view/use_state.rs

```diff
@@ -0,0 +1,62 @@
+#[macro_export]
+macro_rules! generate_use_state_view {
+    ($viewtrait:ident, $cx:ty, $changeflags:ty; $($ss:tt)*) => {
+        pub fn use_state<F, S, G, V>(make_state: F, make_view: G) -> UseState<F, S, G, V> {
+            UseState::new(make_state, make_view)
+        }
+
+        pub struct UseState<F, S, G, V> {
+            make_state: F,
+            state: Option<S>,
+            make_view: G,
+            view: Option<V>,
+        }
+
+        impl<F, S, G, V> UseState<F, S, G, V> {
+            pub fn new(make_state: F, make_view: G) -> Self {
+                Self {
+                    make_state,
+                    state: None,
+                    make_view,
+                    view: None,
+                }
+            }
+        }
+
+        impl<T, A, F, S, G, V> View<T, A> for UseState<F, S, G, V>
+        where
+            S: Send,
+            F: FnMut() -> S + Send,
+            G: FnMut(&mut S) -> V + Send,
+            V: $viewtrait<S, A>,
+        {
+            type State = V::State;
+            type Element = V::Element;
+
+            fn build(&self, cx: &mut $cx) -> (Id, Self::State, Self::Element) {
+                todo!()
```

Comment:
This seems rather important.

Reply:
Lol agree with ya there... Just wanted to make sure I was heading in the right direction. Definitely looks like there are a few directions to go here.

If you're curious this is basically the method implementation I had in mind: [remember](https://github.com/concoct-rs/concoct/blob/900f505718d8551ef277d984d7d0b21e320bdde7/src/view/remember.rs#L33)
Here I use the second function argument to create a view with the state from `Remember`. The first function would create the state only once